src/iuse.cpp

```diff
@@ -6459,20 +6559,800 @@ struct npc_photo_def : public JsonDeserializer, public JsonSerializer {
+                                              name ) );
+                } else {
+                    p->add_msg_if_player( m_bad, string_format(
+                                              _( "But the quality of <color_light_blue>%s</color> image is worse than the previous one." ),
+                                              name ) );
+                }
+            }
+        }
+    }
+    it->set_var( "CAMERA_MONSTER_PHOTOS", monster_photos );
+}
+
+// throws exception
+static bool item_read_extended_photos( item *it, std::vector<extended_photo_def> &extended_photos,
+                                       std::string var_name, bool insert_at_begin )
+{
+    bool result = false;
+    std::istringstream extended_photos_data( it->get_var( var_name ) );
+    JsonIn json( extended_photos_data );
+    if( insert_at_begin ) {
+        std::vector<extended_photo_def> temp_vec;
+        result = json.read( temp_vec );
+        extended_photos.insert( std::begin( extended_photos ), std::begin( temp_vec ),
+                                std::end( temp_vec ) );
+    } else {
+        result = json.read( extended_photos );
+    }
+    return result;
+}
+
+// throws exception
+static void item_write_extended_photos( item *it,
+                                        const std::vector<extended_photo_def> &extended_photos,
+                                        std::string var_name )
+{
+    std::ostringstream extended_photos_data;
+    JsonOut json( extended_photos_data );
+    json.write( extended_photos );
+    it->set_var( var_name, extended_photos_data.str() );
+}
+
+static bool show_photo_selection( player *p, item *it, const std::string &var_name )
+{
+    if( p->is_blind() ) {
+        p->add_msg_if_player( _( "You can't see the camera screen, you're blind." ) );
+        return false;
+    }
+
+    uilist pmenu;
+    pmenu.text = _( "Photos saved on camera:" );
+
+    std::vector<std::string> descriptions;
+    std::vector<extended_photo_def> extended_photos;
+
+    try {
+        item_read_extended_photos( it, extended_photos, var_name );
+    } catch( const JsonError &e ) {
+        debugmsg( "Error reading photos: %s", e.c_str() );
+    }
+    try { // if there is old photos format, append them; delete old and save new
+        if( item_read_extended_photos( it, extended_photos, "CAMERA_NPC_PHOTOS", true ) ) {
+            it->erase_var( "CAMERA_NPC_PHOTOS" );
+            item_write_extended_photos( it, extended_photos, var_name );
+        }
+    } catch( const JsonError &e ) {
+        debugmsg( "Error migrating old photo format (read success = %i): %s",
+                  ( int )!it->has_var( "CAMERA_NPC_PHOTOS" ), e.c_str() );
+    }
+
+    int k = 0;
+    for( const extended_photo_def &extended_photo : extended_photos ) {
+        std::string menu_str = std::regex_replace( extended_photo.name, std::regex( p->name ), _( "You" ) );
```

Comment:
What if my character is named "**" (that would be an invalid regex and would consequently trigger an error here)? What if my name is ".*" (that regex matches *all* text, so the whole text would be replaced with a single "You")?

Reply:
Didn't thought about it, thanks... Such use of regex is really dangerous here.